bindings/java/src/org/sleuthkit/datamodel/CaseDbAccessManager.java

```diff
@@ -602,6 +607,49 @@ public void select(final String sql, final CaseDbAccessQueryCallback queryCallba
 		}
 	}
 	
+
+	/**
+	 * Creates a prepared statement object for the purposes of running a select
+	 * statement.
+	 *
+	 * @param sql The select statement without the starting select keyword.
+	 *
+	 * @return The prepared statement object.
+	 *
+	 * @throws TskCoreException
+	 */
+	@Beta
+	public CasePreparedStatement prepareSelect(String sql) throws TskCoreException {
+		String selectSQL = "SELECT " + sql; // NON-NLS
+		try {
+			CaseDbConnection connection = tskDB.getConnection();
+			PreparedStatement statement = connection.getPreparedStatement(selectSQL, Statement.NO_GENERATED_KEYS);
+			return new CasePreparedStatement(connection, statement);
+		} catch (SQLException ex) {
+			throw new TskCoreException("Error creating select prepared statement.", ex);
```

Comment:
We should make debugging easier by including the SQL that failed to be prepared in the error message.